### DISCUSSION TITLE
What was the selling point to use arroy

### DISCUSSION THREAD

**Discussion by irevoire** [floneum/floneum#151] [General]
Hello team, co-author of arroy here!

I just discovered your project through [this blog post](https://floneum.com/blog/kalosm_0_2), and it’s really awesome to see this kind of project using our vector store 🔥 

I had a few questions about it:
- What were you using before if there was something?
- What was the main argument to switch to arroy over something else?
- Have you seen any perf improvement upon making the switch? 👀

**Comment by ealmloff** [comment]
> Hello team, co-author of arroy here!
> 
> I just discovered your project through [this blog post](https://floneum.com/blog/kalosm_0_2), and it’s really awesome to see this kind of project using our vector store 🔥

Hello, thanks for your work on arroy, it has been great to work with so far!

> I had a few questions about it:
> 
> * What were you using before if there was something?

I was using [instant-distance](https://crates.io/crates/instant-distance/0.6.1) before switching to arroy.

> * What was the main argument to switch to arroy over something else?
> * Have you seen any perf improvement upon making the switch? 👀

The reason I switched to arroy was to make incremental indexing possible. I haven't done any formal benchmarks before and after the switch, but I am hopeful that incremental indexing will speed up workloads with a constantly updating context significantly. For example, [this code](https://github.com/floneum/floneum/blob/master/interfaces/kalosm/examples/live_qa.rs) listens to your environment, transcribes and embeds text and uses arroy to find context that is similar to questions. As the context grows, performing a full rebuild of the index would not be feasible

**Reply by irevoire** [reply]
Niiiice, on my side, I tried to measure the worst-case scenario for the incremental indexing and got a 10x speedup when arroy needed to create a new tree and something like a 30-40x speedup when we didn’t need to create a new tree.
You’ll probably get something even better with a normal use-case instead of the worst case scenario
